lang/java/reef-runtime-local/src/main/java/org/apache/reef/runtime/local/driver/ContainerManager.java

```diff
@@ -144,55 +137,61 @@ public void onNext(final RemoteMessage<ReefServiceProtos.RuntimeErrorProto> valu
     LOG.log(Level.FINE, "Initialized Container Manager with {0} containers", capacity);
   }
 
-  private Collection<String> normalize(final Collection<String> rackNames) {
+  private static Collection<String> normalize(final Collection<String> rackNames) {
     return normalize(rackNames, true);
   }
 
   /**
-   * Normalizes the rack names.
-   *
-   * @param rackNames
-   *          the rack names to normalize
-   * @param validateEnd
-   *          if true, throws an exception if the name ends with ANY (*)
-   * @return a normalized collection
+   * Normalize rack names.
+   * @param rackNames Collection of rack names to normalize.
+   * @param validateEnd If true, throw an exception if the name ends with ANY (*)
+   * @return Collection of normalized rack names.
    */
-  private Collection<String> normalize(final Collection<String> rackNames,
-      final boolean validateEnd) {
+  private static Collection<String> normalize(final Collection<String> rackNames, final boolean validateEnd) {
+
     final List<String> normalizedRackNames = new ArrayList<>(rackNames.size());
-    final Iterator<String> it = rackNames.iterator();
-    while (it.hasNext()) {
-      String rackName = it.next().trim();
+
+    for (String rackName : rackNames) {
+
+      rackName = rackName.trim();
       Validate.notEmpty(rackName, "Rack names cannot be empty");
+
       // should start with a separator
       if (!rackName.startsWith(Constants.RACK_PATH_SEPARATOR)) {
         rackName = Constants.RACK_PATH_SEPARATOR + rackName;
       }
+
       // remove the ending separator
       if (rackName.endsWith(Constants.RACK_PATH_SEPARATOR)) {
         rackName = rackName.substring(0, rackName.length() - 1);
       }
+
       if (validateEnd) {
         Validate.isTrue(!rackName.endsWith(Constants.ANY_RACK));
       }
+
       normalizedRackNames.add(rackName);
     }
+
     return normalizedRackNames;
   }
 
   private void init() {
+
     // evenly distribute the containers among the racks
     // if rack names are not specified, the default rack will be used, so the denominator will always be > 0
-    final int capacityPerRack = capacity / availableRacks.size();
-    int missing = capacity % availableRacks.size();
+    final int capacityPerRack = this.capacity / this.availableRacks.size();
```

Comment:
Does `capacity` here mean actual (i.e., not max theoretically possible) number of containers per rack ?